src/creature_states.c

```diff
@@ -3044,42 +3042,86 @@ short creature_wants_a_home(struct Thing *creatng)
     return 0;
 }
 
-short creature_wants_salary(struct Thing *creatng)
+struct Room* get_room_for_thing_salary(struct Thing* creatng, unsigned char *navtype)
 {
-    SYNCDBG(8,"Starting for %s index %d owner %d", thing_model_name(creatng),(int)creatng->index,(int)creatng->owner);
     TRACE_THING(creatng);
```

Comment:
the TRACE_THING should be placed in functions that make changes to the thing, in creature_wants_salary I see continue_state being set, but in get_room_for_thing_salary I don't see any thing fields being set, so trace thing probably belongs in creature_wants_salary

Reply:
Thanks, moved.